src/main/perf/TaskParser.java

```diff
@@ -327,6 +362,27 @@ Query buildQuery(String type, String text, int minShouldMatch) throws ParseExcep
       if (query.toString().equals("")) {
         throw new RuntimeException("query text \"" + text + "\" parsed to empty query");
       }
+
+      if (combinedFields != null) {
+        CombinedFieldQuery.Builder cfqBuilder = new CombinedFieldQuery.Builder();
+
+        for (FieldAndWeight fieldAndWeight : fieldAndWeights) {
+          cfqBuilder.addField(fieldAndWeight.field, fieldAndWeight.weight);
+        }
+
+        if (query instanceof TermQuery) {
+          cfqBuilder.addTerm(((TermQuery) query).getTerm().bytes());
+        } else if (query instanceof BooleanQuery) {
+          for (BooleanClause clause : (BooleanQuery) query) {
+            cfqBuilder.addTerm(((TermQuery) clause.getQuery()).getTerm().bytes());
+          }
```

Comment:
Let's check that all clauses are SHOULD clauses?

Reply:
Done.